youtube_dl/extractor/twitter.py

```diff
@@ -440,14 +456,19 @@ def _real_extract(self, url):
                 'tweet_mode': 'extended',
             })
 
-        title = description = status['full_text'].replace('\n', ' ')
-        # strip  'https -_t.co_BJYgOjSeGA' junk from filenames
-        title = re.sub(r'\s+(https?://[^ ]+)', '', title)
+        # Get metadata
+        description = status['full_text'].replace('\n', ' ')
         user = status.get('user') or {}
         uploader = user.get('name')
+        uploader_id = user.get('screen_name')
+        timestamp = unified_timestamp(status.get('created_at'))
+
+        # Keep the file name short so it doesn't exceed filesystem limits
+        title = description
+        if len(title) > 40:
+            title = title[:35] + "[...]"
```

Comment:
If you're going to elide the end, I think `...` (not an actual Unicode ellipsis character) will be fine for the elision marker. Also, single quotes.

Looking at these test cases, it could be worth adding a heuristic to break at whitespace, say:
```suggestion
            trim = 35
            # allow up to 10 more characters to find whitespace 
            m = re.search(r'\s+', title, trim, trim + 10)
            if m:
                trim = m.start() + 1
            title = title[:trim] + '...'
```

BTW see #29989 for my general attempt to fix this problem that hasn't quite got off the ground.